components/use-crossposter.js

```diff
@@ -41,41 +117,135 @@ export default function useCrossposter () {
           </Button>
           {' | '}
           <Button
-            variant='link' onClick={() => resolve('skip')}
+            variant='link' onClick={handleSkip}
           >Skip
           </Button>
         </>,
-        () => resolve('skip') // will skip if user closes the toast
+        {
+          onCancel: () => handleSkip()
+        }
       )
     })
   }
 
-  return useCallback(async item => {
+  const crosspostError = (errorMessage) => {
+    return toaster.danger(`Error crossposting: ${errorMessage}`)
+  }
+
+  async function handleEventCreation (item) {
+    const determineItemType = (item) => {
+      const typeMap = {
+        url: 'link',
+        bounty: 'bounty',
+        pollCost: 'poll'
+      }
+
+      for (const [key, type] of Object.entries(typeMap)) {
+        if (item[key]) {
+          return type
+        }
+      }
+
+      // Default
+      return 'discussion'
+    }
+
+    const itemType = determineItemType(item)
+
+    switch (itemType) {
+      case 'discussion':
+        return await discussionToEvent(item)
+      case 'link':
+        return await linkToEvent(item)
+      case 'bounty':
+        return await bountyToEvent(item)
+      case 'poll':
+        return await pollToEvent(item)
+      default:
+        return crosspostError('Unknown item type')
+    }
+  }
+
+  const fetchItemData = async (itemId) => {
+    try {
+      const { data } = await fetchItem({ variables: { id: itemId } })
+
+      return data?.item
+    } catch (e) {
+      console.error(e)
+      return null
+    }
+  }
+
+  const crosspostItem = async item => {
     let failedRelays
     let allSuccessful = false
     let noteId
 
-    do {
-      // XXX we only use discussions right now
-      const event = await discussionToEvent(item)
-      const result = await crosspost(event, failedRelays || relays)
+    const event = await handleEventCreation(item)
+    if (!event) return { allSuccessful, noteId }
 
-      noteId = result.noteId
+    do {
+      try {
+        const result = await crosspost(event, failedRelays || relays)
 
-      failedRelays = result.failedRelays.map(relayObj => relayObj.relay)
+        noteId = result.noteId
+        failedRelays = result?.failedRelays.map(relayObj => relayObj.relay)
```

Comment:
`failedRelays` is undefined if `crosspost` returned `{ error }`. I think that case is not handled here?

Reply:
Initializing failedRelays as an empty array now to handle possibility of it being undefined. Also making sure array is emptied after crosspost.
43b50bf1ba087f8d772e6078237ccc90ddf153e8